### PR TITLE
Don't use mutable global variables in simulator

### PR DESCRIPTION
**Pull request by takluyver**

### CODE REVIEW

File: karabo_bridge/simulation.py [hunk 1]
```diff
@@ -29,23 +29,24 @@
...
-# LPD
-# _PULSES = 32
-# _MODULES = 16
-# _MOD_X = 256
-# _MOD_Y = 256
-# _SHAPE = (_PULSES, _MODULES, _MOD_X, _MOD_Y)
+DETECTORS = {
+    'AGIPD': {
+        'source': 'SPB_DET_AGIPD1M-1/DET/detector',
+        'pulses': 32,
```
Comment (tmichela) [inline karabo_bridge/simulation.py]: we can change this to `64`

File: karabo_bridge/simulation.py [hunk 2]
```diff
@@ -29,23 +29,24 @@
...
+        'modules': 16,
+        'module_shape': (512, 128),  # (x, y)
+    },
+    'LPD': {
+        'source': 'FXE_DET_LPD1M-1/DET/detector',
+        'pulses': 32,
```
Comment (tmichela) [inline karabo_bridge/simulation.py]: and this to `300`

File: karabo_bridge/simulation.py [hunk 3]
```diff
@@ -60,14 +61,15 @@ def gen_combined_detector_data(source, tid_counter, corrected=False, nsources=1)
         'timestamp.frac': frac.ljust(18, '0')  # attosecond resolution
     }
 
-    # detector random data
     if corrected:
-        global _PULSES
-        _PULSES = 31
-        global _SHAPE
-        _SHAPE = (_PULSES, _MODULES, _MOD_X, _MOD_Y)
+        pulse_count = 31   # ???
```
Comment (tmichela) [inline karabo_bridge/simulation.py]: I can explain offline

File: karabo_bridge/simulation.py [hunk 4]
```diff
@@ -60,14 +61,15 @@ def gen_combined_detector_data(source, tid_counter, corrected=False, nsources=1)
...
+    else:
+        pulse_count = detector_info['pulses']
+    array_shape = (pulse_count, detector_info['modules']) + detector_info['module_shape']
 
-        gain_data = np.zeros(_SHAPE, dtype=np.uint16)
+    # detector random data
+    if corrected:
+        gain_data = np.zeros(array_shape, dtype=np.uint16)
         passport = [
             'SPB_DET_AGIPD1M-1/CAL/THRESHOLDING_Q3M2',
```
Comment (tmichela) [inline karabo_bridge/simulation.py]: maybe these can be generated out of data source name, not to confuse people in case using LPD generated data.

Comment (takluyver) [reply karabo_bridge/simulation.py]: Opened #28 for that.

**Pull request merged**